adapters/beachfront/beachfront.go

```diff
@@ -111,300 +113,334 @@ type BeachfrontResponseSlot struct {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		}
+	}
 
-func getEndpoint(request *openrtb.BidRequest) (uri string) {
-	for i := range request.Imp {
-		if request.Imp[i].Video != nil {
-			// If there are any video imps, we will be running a video auction
-			// and dropping all of the banner actions.
-			return VideoEndpoint
+	// At most, I only ever have one banner request, and it does not need the cookie, so doing it first.
+	if len(beachfrontRequests.Banner.Slots) > 0 {
+		bytes, err := json.Marshal(beachfrontRequests.Banner)
+
+		if err == nil {
+			reqs = append(reqs, &adapters.RequestData{
+				Method:  "POST",
+				Uri:     BannerEndpoint,
+				Body:    bytes,
+				Headers: headers,
+			})
+		}
+	}
+
+	if request.User != nil && request.User.BuyerUID != "" {
+		headers.Add("Cookie", "__io_cid="+request.User.BuyerUID)
+	}
+
+	for i := 0; i < len(beachfrontRequests.Video); i++ {
+
+		bytes, err := json.Marshal(beachfrontRequests.Video[i])
+
+		if err == nil {
+			reqs = append(reqs, &adapters.RequestData{
+				Method:  "POST",
+				Uri:     VideoEndpoint + beachfrontRequests.Video[i].AppId + VideoEndpointSuffix,
+				Body:    bytes,
+				Headers: headers,
+			})
+		} else {
+			continue
 		}
 	}
-	return BannerEndpoint
+
+	return reqs, errs
 }
 
-func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var beachfrontRequests BeachfrontRequests
-	var reqJSON []byte
-	var uri string
-	var errs = make([]error, 0, len(request.Imp))
-	var err error
-	var imps int
+func preprocess(request *openrtb.BidRequest) (beachfrontReqs beachfrontRequests, errs []error) {
+	var videoImps = make([]openrtb.Imp, 0)
+	var bannerImps = make([]openrtb.Imp, 0)
 
-	uri = getEndpoint(request)
+	for i := 0; i < len(request.Imp); i++ {
+		if request.Imp[i].Banner != nil {
+			bannerImps = append(bannerImps, request.Imp[i])
+		}
 
-	beachfrontRequests, errs, imps = preprocess(request, uri)
+		if request.Imp[i].Video != nil {
+			videoImps = append(videoImps, request.Imp[i])
+		}
+	}
 
-	if uri == VideoEndpoint {
-		reqJSON, err = json.Marshal(beachfrontRequests.Video)
-		uri = uri + beachfrontRequests.Video.AppId + VideoEndpointSuffix
-	} else {
-		/*
-			We will get here if request contains no Video imps, though it might have
-			Audio or Native imps as well as banner.
-		*/
-		reqJSON, err = json.Marshal(beachfrontRequests.Banner)
+	if len(bannerImps) == 0 && len(videoImps) == 0 {
+		errs = append(errs, errors.New("no valid impressions were found in the request"))
+		return
 	}
 
-	if imps == 0 {
-		errs = append(errs, errors.New("no valid impressions were found"))
-		return nil, errs
+	if len(bannerImps) > 0 {
+		request.Imp = bannerImps
+		beachfrontReqs.Banner, errs = getBannerRequest(request)
 	}
 
-	if err != nil {
-		errs = append(errs, err)
-		return nil, errs
+	if len(videoImps) > 0 {
+		request.Imp = videoImps
+
+		if len(errs) > 0 {
```

Comment:
You can append a slice to another slice by doing `slice2 = append(slice2, slice1...)`. The `...` after the second argument to the `append` call lets you pass in a slice versus a single value. And so, you can replace this `if else` block with:

```
beachfrontReqs.Video, videoErrs = getVideoRequests(request)
errs = append(errs, videoErrs...)
```

This will append error returned by the `getVideoRequests` method call to the `errs` slice which should already have the errors returned by the `getBannerRequest` method call above, if any.